bot/on_message/bots/openai_bot.py

```diff
@@ -112,63 +128,59 @@ def should_query_weezerpedia_api(self, last_three_messages):
             print(f"An error occurred: {e}")
             return None
 
-    def fetch_openai_completion(self, message, system, incoming_message_text):
 
-        system_message = {"role": "system", "content": system}
+    def fetch_openai_completion(self, prompt_params: PromptParams, num_messages_lookback: int):
+        system_message = {"role": "system", "content": prompt_params.system_prompt}
 
-        if message.channel.id not in self.openai_sessions:
-            self.openai_sessions[message.channel.id] = []
+        if prompt_params.channel_id not in self.openai_sessions:
+            self.openai_sessions[prompt_params.channel_id] = []
 
-        messages_in_this_channel = self.openai_sessions[message.channel.id]
+        messages_in_this_channel = self.openai_sessions[prompt_params.channel_id]
 
         # Remove any existing system messages
         messages_in_this_channel = [
-            msg for msg in messages_in_this_channel if msg['role'] != 'system']
-
-        # Update the session with messages excluding old system messages
-        self.openai_sessions[message.channel.id] = messages_in_this_channel
+            msg for msg in messages_in_this_channel if msg["role"] != "system" or "[INTERNAL]" not in msg["content"]]
 
         # Add the new system message at the beginning
         new_content = [system_message] + messages_in_this_channel
 
-        # # Step 2: Add any context from Weezerpedia API if needed
+        # Replace the channel messages with the cleaned up content
+        self.openai_sessions[prompt_params.channel_id] = new_content
```

Comment:
i cleaned this up a bit. the new_content is mutable and passed by reference. so modifying it later will modify it here.

side note: there might be a case to use discord bot's channel history function if we think theres any value in the deleted messages.